include/boost/geometry/algorithms/detail/disjoint/point_point.hpp

```diff
@@ -40,12 +48,103 @@ namespace boost { namespace geometry
 namespace detail { namespace disjoint
 {
 
+
+class point_point_on_spheroid
+{
+private:
+    template
+    <
+        typename Units,
+        typename CoordinateType1,
+        typename CoordinateType2
+    >
+    static inline bool apply_same_units(CoordinateType1& lon1,
+                                        CoordinateType1& lat1,
+                                        CoordinateType2& lon2,
+                                        CoordinateType2& lat2)
+    {
+        math::normalize_spheroidal_coordinates<Units>(lon1, lat1);
+        math::normalize_spheroidal_coordinates<Units>(lon2, lat2);
+
+        return ! math::equals(lat1, lat2) || ! math::equals(lon1, lon2);
+    }
+
+public:
+    template <typename Point1, typename Point2>
+    static inline bool apply(Point1 const& p1, Point2 const& p2)
+    {
+        typedef typename coordinate_type<Point1>::type coordinate_type1;
+        typedef typename coordinate_type<Point2>::type coordinate_type2;
+
+        typedef typename coordinate_system<Point1>::type::units units1;
+        typedef typename coordinate_system<Point2>::type::units units2;
+
+        if (BOOST_GEOMETRY_CONDITION((boost::is_same<units1, units2>::value)))
+        {
+            coordinate_type1 lon1 = geometry::get<0>(p1);
+            coordinate_type1 lat1 = geometry::get<1>(p1);
+            coordinate_type2 lon2 = geometry::get<0>(p2);
+            coordinate_type2 lat2 = geometry::get<1>(p2);
+
+            return apply_same_units<units1>(lon1, lat1, lon2, lat2);
+        }
+
+        typedef typename geometry::select_most_precise
+            <
+                typename fp_coordinate_type<Point1>::type,
+                typename fp_coordinate_type<Point2>::type
+            >::type calculation_type;
+
+        calculation_type lon1 = get_as_radian<0>(p1);
+        calculation_type lat1 = get_as_radian<1>(p1);
+
+        calculation_type lon2 = get_as_radian<0>(p2);
+        calculation_type lat2 = get_as_radian<1>(p2);
+
+        return apply_same_units<radian>(lon1, lat1, lon2, lat2);
+    }
+};
+
+
+template
+<
+    typename Point1, typename Point2,
+    std::size_t Dimension, std::size_t DimensionCount,
+    typename CSTag1 = typename cs_tag<Point1>::type,
+    typename CSTag2 = CSTag1
+>
+struct point_point
+    : point_point<Point1, Point2, Dimension, DimensionCount, cartesian_tag>
+{};
+
```

Comment:
We often used strategies for this (to distinguish between coordinate systems). 
But I think that dispatching it here is perfectly fine here.